src/j1client.js

```diff
@@ -90,20 +103,41 @@ class JupiterOneClient {
     let results = [];
 
     while (!complete) {
-      const query = gql`
-      {
-        queryV1(query: "${j1ql} SKIP ${page *
-        J1QL_SKIP_COUNT} LIMIT ${J1QL_LIMIT_COUNT}") {
-          data
-        }
-      }`;
+      const query = queryV1Gql(j1ql, page);
       page++;
 
       const res = await this.graphClient.query({ query });
       if (res.errors) {
         throw new Error(`JupiterOne returned error(s) for query: '${j1ql}'`);
       }
-      const { data } = res.data.queryV1;
+
+      const deferredUrl = res.data.queryV1.url;
+      let status = JobStatus.IN_PROGRESS;
+      let statusFile;
+      let startTimeInMs = Date.now();
+      do {
+        await sleep(200);
+        statusFile = await fetch(deferredUrl).then(res => res.json());
+        status = statusFile.status;
+        if (Date.now() - startTimeInMs > QUERY_RESULTS_TIMEOUT) {
```

Comment:
Should we skip the check here if the status in the line above changes to `COMPLETE`?

Reply:
@charlieduong94 I'll do the `while` check directly after getting the status

Reply:
Moved `while` clause to directly after getting `status`